### PR TITLE
Test main function with unit tests with files separation

### PR DESCRIPTION
**Pull request by Czaki**
@YannickJadoul  I separate and left old test in `platform_test.py` file.

### CODE REVIEW

**Comment by YannickJadoul** [comment]
@Czaki Thanks; good to compare with #232!

There's still more noise than I expected, though, so maybe #232 is equally good? I find it hard to judge, actually; sorry :-/

File: unit_test/main_function_test.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+def test_build_selector(system, build, build_set, skip, skip_set, monkeypatch, argtest):
+    apply_mock_protection(monkeypatch)
+    monkeypatch.setattr(globals()[system], "build", argtest)
+    env = {"CIBW_PLATFORM": system, "CIBW_BUILD": build, "CIBW_SKIP": skip}
+    monkeypatch.setattr(os, 'environ', env)
+    main()
+    assert isinstance(argtest.kwargs["build_selector"], BuildSelector)
+    selector = argtest.kwargs["build_selector"]
+    selected = set([x for x in ["aaaa", "abcd", "abab", "bcde"] if selector(x)])
+    assert selected == build_set.difference(skip_set)
```
Comment (YannickJadoul) [inline unit_test/main_function_test.py]: We already have `build_selector_test.py` for this, for example, and integration tests to see if `CIBW_BUILD`/`CIBW_SKIP` work. Do we need this here again?

Comment (Czaki) [reply unit_test/main_function_test.py]: I'm not sure. This check it faster, but of course there is redundancy.

**Comment by YannickJadoul** [comment]
Closing in favour of #232

**Pull request closed without merging**